quickwit-metastore/src/metastore/index_metadata.rs

```diff
@@ -0,0 +1,414 @@
+                    "type": "i64",
+                    "fast": true
+                },
+                {
+                    "name": "body",
+                    "type": "text",
+                    "stored": true
+                },
+                {
+                    "name": "response_date",
+                    "type": "date",
+                    "fast": true
+                },
+                {
+                    "name": "response_time",
+                    "type": "f64",
+                    "fast": true
+                },
+                {
+                    "name": "response_payload",
+                    "type": "bytes",
+                    "fast": true
+                },
+                {
+                    "name": "owner",
+                    "type": "text",
+                    "tokenizer": "raw"
+                },
+                {
+                    "name": "attributes",
+                    "type": "object",
+                    "field_mappings": [
+                        {
+                            "name": "tags",
+                            "type": "array<i64>"
+                        },
+                        {
+                            "name": "server",
+                            "type": "text"
+                        },
+                        {
+                            "name": "server.status",
+                            "type": "array<text>"
+                        },
+                        {
+                            "name": "server.payload",
+                            "type": "array<bytes>"
+                        }
+                    ]
+                }
+            ],
+            "tag_fields": ["owner"],
+            "store_source": true
+        }"#;
+        let doc_mapping = serde_json::from_str(doc_mapping_json).unwrap();
+        let indexing_settings = IndexingSettings {
+            timestamp_field: Some("timestamp".to_string()),
+            sort_field: Some("timestamp".to_string()),
+            sort_order: Some(SortOrder::Desc),
+            resources: IndexingResources::for_test(),
+            ..Default::default()
+        };
+        let search_settings = SearchSettings {
+            default_search_fields: vec![
+                "body".to_string(),
+                "attributes.server".to_string(),
+                "attributes.server.status".to_string(),
+            ],
+        };
+        let now_timestamp = Utc::now().timestamp();
+        Self {
+            index_id: index_id.to_string(),
+            index_uri: index_uri.to_string(),
+            source_checkpoints: Default::default(),
+            doc_mapping,
+            indexing_settings,
+            search_settings,
+            sources: Vec::new(),
+            create_timestamp: now_timestamp,
+            update_timestamp: now_timestamp,
+        }
+    }
+
+    /// Returns the checkpoint associated with a source.
+    pub fn get_checkpoint(&self, source_id: &str) -> Checkpoint {
```

Comment:
I think it would be better to insert an empty source checkpoint when creating an index or adding a new source and fail here if the `source_id` does not exist.

Reply:
Maybe that doesn't work for ad-hoc sources for the ingest command.
t_t

Reply:
Maybe a source config should come with an "init checkpoint". That would also be useful for starting for a particular point for a Kafka source or prefix for an S3 source.

Reply:
Opened #991